src/main.js

```diff
@@ -1,12 +1,21 @@
-import cli from 'cli';
 import Linter from 'linter';
 import log from 'logger';
 
 import 'babel-polyfill';
 
+// Only include the CLI module if this is being run as a command line script.
+// Otherwise we trample over other apps using yargs. See:
+// https://github.com/mozilla/addons-linter/issues/735
+var cli = {argv: {logLevel: 'fatal'}};
```

Comment:
FWIW, I tried adding [`"no-var": "warn"`](http://eslint.org/docs/rules/no-var) to the .eslintrc and then running `$ grunt eslint` and got:

> ✖ 916 problems (0 errors, 916 warnings)

Similarly, if I add [`"prefer-const": "warn"`](http://eslint.org/docs/rules/prefer-const), I get:

> ✖ 91 problems (0 errors, 91 warnings)

The good news is that both those rules seem auto-fixable by newer ESLint versions, so the change isn't that difficult to quickly make. The bad news is, it's an ugly diff and touches virtually every file (plus we'll have a few errors with the `max-len` rule, since changing from `var` to `const` adds 2 characters, which means you'll either need to re-break some lines to be shorter, or tweak the `max-len` rule to accept 82 char lines.